ultralytics/utils/tlc/engine/dataset.py

```diff
@@ -7,48 +7,20 @@
 
 from ultralytics.utils import LOGGER
 
-# Responsible for any generic 3LC dataset handling, such as using sampling weights
-# Assume there is an attribute self.table that is a tlc.Table, and self._example_ids
+# Responsible for any generic 3LC dataset handling, such as scanning, caching and adding example ids to each sample
+# Assume there is an attribute self.table that is a tlc.Table
 class TLCDatasetMixin:
-    def _post_init(self, sampling_weights=False):
+    def _post_init(self):
         self.display_name = self.table.dataset_name
-        # Checks
-        if sampling_weights and tlc.SAMPLE_WEIGHT not in self.table.table_rows[0]:
-            raise ValueError("Cannot use sampling weights with no sample weight column in the table.")
 
         assert hasattr(self, "table") and isinstance(self.table, tlc.Table), "TLCDatasetMixin requires an attribute `table` which is a tlc.Table."
-        # Assume instance has self._example_ids (index -> example_id mapping)
-
-        if not hasattr(self, "_indices"):
-            # Sequentially iterate over the samples by default
-            self._indices = np.arange(len(self.example_ids))
-
-        sample_weights = [
-            self.table.table_rows[example_id][tlc.SAMPLE_WEIGHT]
-            for example_id in self.example_ids
-        ]
-        self._sample_probabilities = np.array(sample_weights) / np.sum(sample_weights)
-
-    def resample_indices(self):
-        # Sample from available indices
-        indices = np.arange(len(self.example_ids))
-        self._indices[:] = np.random.choice(indices, len(indices), p=self._sample_probabilities)
+        if not hasattr(self, "example_ids"):
+            self.example_ids = np.arange(len(self.table.table_rows))
```

Comment:
Any reason for calling len(table_rows) and not len(table)?

Reply:
Not at all, fixing!